src/command.rs

```diff
@@ -91,6 +93,24 @@ pub struct Examples {
     pub dir_name: Option<String>,
 }
 
+#[derive(Debug)]
+pub enum Submission {
+    Id(u64),
+    NthLast(String, u64),
+}
+impl Submission {
+    fn parse(pargs: &mut pico_args::Arguments) -> Result<Submission> {
+        let task: Option<String> = pargs.opt_value_from_str(["-t", "--task"])?;
+        let submission_id: Option<u64> = pargs.opt_free_from_str()?;
+
+        Ok(if let Some(task) = task {
+            Submission::NthLast(task, submission_id.unwrap_or(1))
+        } else {
+            Submission::Id(submission_id.ok_or_else(|| anyhow!("Submission ID not specified"))?)
+        })
+    }
+}
+
```

Comment:
This could be moved a little bit lower where the other `impl`s are. Also `pargs.opt_value_from_str(["-t", "--task"])?` --> `parse_task_id(pargs)?`.

Reply:
Should we use zero indexing instead? Either way is fine but I have a memory that we ended up choosing the zero indexing in some meeting.

Reply:
@kluopaja Yeah, that was decided after this PR though

Reply:
@ollpu I see. I switched it to zero indexing now since the ui/service branch depends on this.